src/components/base/BaseCooperationPoll.vue

```diff
@@ -9,36 +25,148 @@
 
       <div class="poll-field">
         <span>Дата початку:</span>
-        <div>{{ poll.creationDate }}</div>
+        <div>{{ poll.creationDate.toLocaleString('uk-UA') }}</div>
       </div>
 
       <div class="poll-field">
         <span>Дата завершення:</span>
-        <div>{{ poll.completionDate }}</div>
+        <div>{{ poll.completionDate.toLocaleString('uk-UA') }}</div>
       </div>
     </div>
   </article>
+
+  <Dialog
+    header="Редагувати опитування"
+    v-model:visible="displayEditPollModal"
+    :style="{ width: '580px' }"
+    :modal="true"
+    :closable="false"
+    :dismissableMask="true"
+  >
+    <EditCooperationPollForm
+      :poll="$props.poll"
+      :displayEditPollModal="displayEditPollModal"
+      @close-edit-poll="displayEditPollModal = false"
+      :showSuccessOperation="showSuccessOperation"
+    ></EditCooperationPollForm>
+  </Dialog>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
+import { mapGetters } from 'vuex';
 import { PollModel } from '@/store/polls/models/poll.model';
+import Button from 'primevue/button';
+import Menu from 'primevue/menu';
+import { StoreModuleEnum } from '@/store/types';
+import { PollsActionEnum } from '@/store/polls/types';
+import { CooperationGettersEnum } from '@/store/cooperation/types';
+import ConfirmPopup from 'primevue/confirmpopup';
+import Dialog from 'primevue/dialog';
+import EditCooperationPollForm from '@/components/EditCooperationPollForm.vue';
 
 export default defineComponent({
   name: 'BaseCooperationPoll',
+  components: {
+    Button,
+    Menu,
+    ConfirmPopup,
+    Dialog,
+    EditCooperationPollForm,
+  },
   props: {
     poll: {
       type: PollModel,
       required: true,
     },
   },
+  data() {
+    return {
+      pollActions: () => {
+        return [
+          {
+            label: 'Видалити опитування',
+            icon: 'pi pi-times',
+            type: 'Submit',
+            command: () => {
+              this.confirmDeletePoll(event);
+            },
+          },
+          {
+            label: 'Редагувати опитування',
+            icon: 'pi pi-user-edit',
+            command: () => {
+              this.openEditPollModal();
+            },
+          },
+        ];
+      },
+      displayEditPollModal: false,
+    };
+  },
   methods: {
     openPageWithPollInfo(id: any) {
       this.$router.push({
         name: 'poll-info',
         params: { id: id },
       });
     },
+    checkStatus(status: string): boolean {
+      if (status !== 'draft') {
+        this.showFailOperation('видалено або редаговано');
+        return true;
+      }
+      return false;
+    },
+    toggle(event: any): void {
+      event.stopPropagation();
+      (this.$refs.menu as any).toggle(event);
+    },
```

Comment:
try to avoid any type

```suggestion
    toggle(event: Event): void { // | KeyboardEvent | string | boolean whatever can be an event
      event.stopPropagation();
      (this.$refs.menu as Vue & { toggle: (event: Event) => void }).toggle(event); // or return type of toggle
    },
```

Reply:
I'm not able to get this part of code (this.$refs.menu as Vue & { toggle: (event: Event) => void }). I didn't find any information about typing refs

Reply:
refs object returns Vue component or HTML element
In this case it's Vue component
Default Vue component is an object, which does not have `toggle` method
So you have to add & { ... } to assert existing of this method

Reply:
Well, thanks)